sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala

```diff
@@ -688,6 +688,36 @@ abstract class SparkPlanMeta[INPUT <: SparkPlan](plan: INPUT,
     // shuffled exchanges. So broadcast exchanges are not impacted which could have an impact on
     // BroadcastHashJoin, and shuffled exchanges are not used to disable anything downstream.
     fixUpExchangeOverhead()
+
+    // 3) Some child nodes can't run on GPU if parent nodes can't run on GPU.
+    // WriteFilesExec is a new operator from Spark version 340,
+    // Did not extract a shim code for simplicity
+    tagChildAccordingToParent(this.asInstanceOf[SparkPlanMeta[SparkPlan]], "WriteFilesExec")
```

Comment:
Not sure this is necessary. But for `WriteFilesExec`, we can override the `tagPlanForGpu` method in the `GpuWriteFilesMeta` to do this tagging.

Reply:
`WriteFilesExec` is more like a flag and does not have input and output, and it's a physical node.
`InsertIntoHadoopFsRelationCommandMeta.tagSelfForGpu` already does what we want, no need to do this again.
`InsertIntoHadoopFsRelationCommandMeta` already checks against the logic plan `InsertIntoHadoopFsRelationCommand.query`, no need to check the physical plan again.

Reply:
I mean doing the check in `GpuWriteFilesMeta` which is only specific to `WriteFilesExec`. 

```
class   GpuWriteFilesMeta {

    override def tagPlanForGpu(): Unit = {
      assert(this.parent.isDefined)
      if (!this.parent.get.canThisBeReplaced) {
        // parent can't run on GPU, also tag this.
        p.willNotWorkOnGpu(
          "WriteFilesExec can't run on GPU because the parent can't run on GPU")
      }
    }
}
```

Reply:
I've tried what you mentioned. The tag logic is a hierarchy, it first tags children nodes and then tags self, this means we can't get the parent tag result when tagging `GpuWriteFilesMeta`.

Reply:
I see. Thx.
So we can create a new sub meta class (e.g. SparkPlanFlagNodeMeta), and override the `runAfterTagRules` method to do this tagging. Then there is no need to hard code the `WriteFilesExec` in an abstract class's method.
```
abstract class SparkPlanFlagNodeMeta extends SparkPlanMeta {
    // default impl
    override def runAfterTagRules(): Unit = {
      super.runAfterTagRules()
      assert(this.parent.isDefined)
      if (!this.parent.get.canThisBeReplaced) {
        // parent can't run on GPU, also tag this.
        willNotWorkOnGpu(
          s"${wrapped.getClass.getSimpleName} can't run on GPU because the parent can't run on GPU")
      }
    }
}
```